### ISSUE TITLE
Feedback on lost device

### ISSUE THREAD

**Issue by gillesdegottex** [thestk/rtaudio#268]
Issues #194 and #195 already speak about disconnection of audio device while being use.

I couldn't find much feedback in an RtAudio stream about this situation and this is likely to happen more often in the future as many headset are now USB, with embedded audio cards.

I've found two ways to get feedback about a stream: through the RtAudioErrorCallback function (called at any point during the API-specific callback) and through the RtAudioStreamStatus (available on each RtAudioCallback calls, which is called once early in the API-specific callback).

Unfortunately:
* RtAudioErrorCallback always get a RtAudioError::WARNING (according to #195 point 2.). A temporary XRUN cannot thus be distinguished from a permanent lost device (the error type is sort of confused with the mean to report the error, because a non-warning leads to exceptions and a warning leads to text output report)
* RtAudioStreamStatus gives information about XRUNs only.

Did I missed any other information somewhere ?

I've made a workaround that is more like a dirty fix in #267

**Comment by garyscavone** [comment]
Based on my work on #195, I started to develop the noexceptions branch but that was over a year ago. @radarsat more recently merged most of the changes from noexceptions into master but I think it may not have been complete, as I still see a “throw” in the RtApi::error() function. Take a look at the noexceptions branch and perhaps submit a PR to update the error function.

When I was working on the “unplugging” issue, I only remember making a fix for the OS-X API. It is possible I also did it for another 1 or 2 APIs. The desired behaviour if someone unplugs a device while a stream is running is to automatically stop the stream and issue an error message (but not throw an exception).
